passthroughserver/main.cpp

```diff
@@ -23,63 +23,25 @@
 #include "waylandplatform.h"
 #endif // HAVE_WAYLAND
 #include "unknownplatform.h"
+#include "logging.h"
 
 #include <QGuiApplication>
 #include <QtDebug>
 
 namespace {
+QLoggingCategory::CategoryFilter defaultLoggingFilter;
 
-void disableMInputContextPlugin()
+void loggingCategoryFilter(QLoggingCategory *category)
 {
-    // none is a special value for QT_IM_MODULE, which disables loading of any
-    // input method module in Qt 5.
-    setenv("QT_IM_MODULE", "none", true);
-}
-
-bool isDebugEnabled()
-{
-    static int debugEnabled = -1;
-
-    if (debugEnabled == -1) {
+    if (qstrcmp(category->categoryName(), "org.maliit.framework") == 0) {
         QByteArray debugEnvVar = qgetenv("MALIIT_DEBUG");
+        bool enabled = false;
         if (!debugEnvVar.isEmpty() && debugEnvVar != "0") {
-            debugEnabled = 1;
-        } else {
-            debugEnabled = 0;
+            enabled = true;
         }
-    }
-
-    return debugEnabled == 1;
-}
-
-void outputMessagesToStdErr(QtMsgType type,
-                            const QMessageLogContext &context,
-                            const QString &msg)
-{
-    Q_UNUSED(context);
-    QByteArray utf_text(msg.toUtf8());
-    const char *raw(utf_text.constData());
-
-    switch (type) {
-    case QtDebugMsg:
-        if (isDebugEnabled()) {
-            fprintf(stderr, "DEBUG: %s\n", raw);
-        }
-        break;
-#if QT_VERSION >= QT_VERSION_CHECK(5, 5, 0)
-    case QtInfoMsg:
-        fprintf(stderr, "INFO: %s\n", raw);
-        break;
-#endif
-    case QtWarningMsg:
-        fprintf(stderr, "WARNING: %s\n", raw);
-        break;
-    case QtCriticalMsg:
-        fprintf(stderr, "CRITICAL: %s\n", raw);
-        break;
-    case QtFatalMsg:
-        fprintf(stderr, "FATAL: %s\n", raw);
-        abort();
+        category->setEnabled(QtDebugMsg, enabled);
```

Comment:
As the default level is Warning as set by the declaration of the category, can't we drop the extra `bool enabled` usage here, and just always all this with true if the `debugEnvVar` check is true?

Reply:
Just doing that wouldn't work because by default all logging is turned on and the default filter would be setting it off here. 

But instead moved now the default handler always done first and the debug override happening later and like suggested to only set debug true. Makes QT_LOGGING_RULES honored better.